polkadot/xcm/procedural/src/builder_pattern.rs

```diff
@@ -103,13 +105,168 @@ fn generate_methods_for_enum(name: syn::Ident, data_enum: &syn::DataEnum) -> Tok
 		}
 	});
 	let output = quote! {
-		impl<Call> XcmBuilder<Call> {
+		impl<Call> XcmBuilderRaw<Call> {
 			#(#methods)*
 
+			/// Create an instance of this type with some pre-filled instructions.
+			/// Useful to create additional builders with some restrictions beforehand.
+			pub(crate) fn with_instructions(instructions: Vec<#name<Call>>) -> Self {
+				Self(instructions)
+			}
+
 			pub fn build(self) -> Xcm<Call> {
 				Xcm(self.0)
 			}
 		}
 	};
 	output
 }
+
+/// All instructions that load the holding register
+const LOAD_HOLDING_INSTRUCTIONS: &[&str] = &[
```

Comment:
Is it not possible to auto-generate this? Otherwise, this will be another thing we will need to keep track of.

Reply:
I couldn't find a way, since these are instructions that we know behave in a certain way. I agree it is another thing to keep in mind when adding or removing instructions and that's a bummer.

Actually, one thing we could do is add an attribute on top of those enum variants, something like `#[loads_holding]`. Then, we wouldn't need to maintain this list separately.